### ISSUE TITLE
Have resource server use its default_file_mode configuration value when creating local replicas

### ISSUE THREAD

**Issue by tedgin** [irods/irods#6712]
## Feature

This is related to #3231.  One use case of iRODS is to colocate a data consuming service on the resource server hosting the data for this service. The service needs file system level permission to access the data, which can be configured using `default_file_mode`.  Currently, iRODS chooses the value of `default_file_mode` set on the iRODS server the uploading client connects to, ii.e., the client's `irods_host` configuration value.  If the client connects to the zone's canonical iRODS host, which may be a load balancer, it is likely that the client won't connect to the colocated resource server, and this server's `default_file_mode` won't be chosen. To prevent this from happening, the `default_file_mode` on all of the iRODS servers needs to be set to the value required on the colocated one.  This isn't intuitive, and it's not always desirable.  

Could iRODS be changed so that the selected `default_file_mode` come from the resource server hosting the storage resource chosen for a new replica?  Or maybe the `unixfilesystem` resource could be modified to accept file mode as a context value.

**Comment by trel** [comment]
Makes sense to me - should honor the policy of the machine writing the file ... after any redirections.

Any downsides we can think of?  Implications we're not thinking of if we change this behavior?

Maybe this should be labeled a bug instead?
